packages/opentelemetry-core/test/correlation-context/HttpCorrelationContext.test.ts

```diff
@@ -0,0 +1,172 @@
+/*!
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  defaultGetter,
+  defaultSetter,
+  CorrelationContext,
+} from '@opentelemetry/api';
+import { Context } from '@opentelemetry/context-base';
+import * as assert from 'assert';
+
+import {
+  getCorrelationContext,
+  setCorrelationContext,
+} from '../../src/correlation-context/correlation-context';
+
+import {
+  HttpCorrelationContext,
+  CORRELATION_CONTEXT_HEADER,
+  MAX_PER_NAME_VALUE_PAIRS,
+} from '../../src/correlation-context/propagation/HttpCorrelationContext';
+
+describe('HttpCorrelationContext', () => {
+  const httpTraceContext = new HttpCorrelationContext();
+
+  let carrier: { [key: string]: unknown };
+
+  beforeEach(() => {
+    carrier = {};
+  });
+
+  describe('.inject()', () => {
+    it('should set traceparent header', () => {
+      const correlationContext: CorrelationContext = {
+        key1: { value: 'd4cda95b652f4a1592b449d5929fda1b' },
+        key3: { value: 'c88815a7-0fa9-4d95-a1f1-cdccce3c5c2a' },
```

Comment:
Could you pls add tests for non-string value?

Reply:
Is this gonna support non-string values? I asking because the `CorrelationContext` interface is defined as a `[key]:EntryValue (1)` and `EntryValue` is defined with a value property as string type (2).

1) https://github.com/open-telemetry/opentelemetry-js/blob/master/packages/opentelemetry-api/src/correlation_context/CorrelationContext.ts

2) https://github.com/open-telemetry/opentelemetry-js/blob/master/packages/opentelemetry-api/src/correlation_context/EntryValue.ts

Reply:
Only string values are allowed, pls ignore my comment.